x/pdv/ante/account.go

```diff
@@ -0,0 +1,36 @@
+package ante
+
+import (
+	"fmt"
+
+	"github.com/Decentr-net/decentr/x/token"
+	"github.com/Decentr-net/decentr/x/utils"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+)
+
+type CreateAccountDecorator struct {
+	tokenKeeper   token.Keeper
+	accountKeeper keeper.AccountKeeper
+}
+
+func NewCreateAccountDecorator(ak auth.AccountKeeper, tokenKeeper token.Keeper) CreateAccountDecorator {
+	return CreateAccountDecorator{
+		tokenKeeper: tokenKeeper, accountKeeper: ak}
+}
+
+func (cad CreateAccountDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
```

Comment:
@mikluke this works, but I have no idea why it is executed twice when a send transaction is submitted.
```
decentrcli tx send $(decentrcli keys show jack -a) $(decentrcli keys show alice -a) 1000udec
```
There are two messaged in console
```
I[2021-04-24|15:15:36.341] account decentr1ztqjzj6h52f23ldwfpvwplqz4gjcc2q4av839v created module=main 
I[2021-04-24|15:15:37.904] account decentr1ztqjzj6h52f23ldwfpvwplqz4gjcc2q4av839v created module=main 
```

Do you have any idea?

Reply:
@mikluke  They have different context. First one checkTx = true, the second - false.